File: hw/rojas/u1/hw11ChickenCoops/chickenFarm.js
// chickenFarm.js
// Owner/Author: Josue Rojas

import fs from "node:fs";
import { Chicken } from "./Chicken.js";
import { ChickenCoop } from "./ChickenCoop.js";

console.log("===  Chicken Farm Simulator ===");

const coop1 = new ChickenCoop(1, "North Coop");
const coop2 = new ChickenCoop(2, "South Coop");


const chickens = [
  new Chicken(1, "Lucy", "White", 2, false),
  new Chicken(2, "Maruja", "Brown", 1, true),
  new Chicken(3, "Coco", "Black", 3, false),
  new Chicken(4, "Bella", "Yellow", 1, false),
  new Chicken(5, "Lola", "Gray", 2, true),
  new Chicken(6, "Nina", "White", 4, false),
  new Chicken(7, "Kira", "Brown", 2, false),
  new Chicken(8, "Daisy", "White", 1, false),
  new Chicken(9, "Molly", "Black", 3, true),
  new Chicken(10, "Gigi", "White", 2, false),
];

chickens.forEach((ch, i) => (i < 6 ? coop1 : coop2).addChicken(ch));

let eggId = 1, poopId = 1;
[coop1, coop2].forEach(coop => {
  coop.chickens.forEach(ch => {
    if (!ch.isMolting && Math.random() < 0.6) coop.collectEgg(ch.layEgg(eggId++));
    coop.collectPoop(ch.poop(poopId++));
  });
});


const showCoopDetails = (coop) => {
  console.log(`\n Coop ID: ${coop.id} | Name: ${coop.name}`);
  console.log(`Chickens (${coop.chickens.length} total):`);
  coop.chickens.forEach(c => {
    console.log(` → id: ${c.id}, name: ${c.name}, color: ${c.color}, age: ${c.age}, isMolting: ${c.isMolting}`);
  });
  console.log(`Eggs collected: ${coop.eggs.length}`);
  console.log(`Poops collected: ${coop.poops.length}`);
};

showCoopDetails(coop1);
showCoopDetails(coop2);

const chickensCSV = [["coopId","coopName","chickenId","name","color","age","isMolting"]];
[coop1, coop2].forEach(coop => {
  coop.chickens.forEach(c =>
    chickensCSV.push([coop.id, coop.name, c.id, c.name, c.color, c.age, c.isMolting])
  );
});
fs.writeFileSync("chickens.csv", chickensCSV.map(r => r.join(",")).join("\n"));

console.log("\n CSV exported: chickens.csv");

File: hw/rojas/u1/hw11ChickenCoops/chicken_farm.py
# chicken_farm.py
# Owner/Author: Josue Rojas

from chicken import Chicken
from chicken_coop import ChickenCoop
import csv
import random

print("===  Chicken Farm Simulator (Python) ===")


coop1 = ChickenCoop(1, "North Coop")
coop2 = ChickenCoop(2, "South Coop")

chickens = [
    Chicken(1, "Lucy", "White", 2, False),
    Chicken(2, "Maruja", "Brown", 1, True),
    Chicken(3, "Coco", "Black", 3, False),
    Chicken(4, "Bella", "Yellow", 1, False),
    Chicken(5, "Lola", "Gray", 2, True),
    Chicken(6, "Nina", "White", 4, False),
    Chicken(7, "Kira", "Brown", 2, False),
    Chicken(8, "Daisy", "White", 1, False),
    Chicken(9, "Molly", "Black", 3, True),
    Chicken(10, "Gigi", "White", 2, False),
]

for i, chicken01 in enumerate(chickens):
    (coop1 if i < 6 else coop2).add_chicken(chicken01)

egg_id, poop_id = 1, 1
for coop in [coop1, coop2]:
    for ch in coop.chickens:
        if not ch.is_molting and random.random() < 0.6:
            coop.collect_egg(ch.lay_egg(egg_id))
            egg_id += 1
        coop.collect_poop(ch.poop(poop_id))
        poop_id += 1

coop1.show_info()
coop2.show_info()

with open("chickens.csv", "w", newline="") as f:
    writer = csv.writer(f)
    writer.writerow(["coopId", "coopName", "chickenId", "name", "color", "age", "isMolting"])
    for coop in [coop1, coop2]:
        for c in coop.chickens:
            writer.writerow([coop.id, coop.name, c.id, c.name, c.color, c.age, c.is_molting])

print("\n CSV file created: chickens.csv")
